File: dkplayer-players/exo/src/main/java/xyz/doikki/videoplayer/exo/CacheConfig.java
package xyz.doikki.videoplayer.exo;



import androidx.annotation.Nullable;

import java.io.File;

public class CacheConfig {

    public static final long DEFAULT_CACHE_MAX_BYTES = 512L * 1024 * 1024;

    public final boolean useBuiltInCache;
    @Nullable
    public final File cacheDir;
    public final long cacheMaxBytes;
    @Nullable
    public final CacheKeyResolver cacheKeyResolver;

    private CacheConfig(Builder b) {
        this.useBuiltInCache = b.useBuiltInCache;
        this.cacheDir = b.cacheDir;
        this.cacheMaxBytes = b.cacheMaxBytes;
        this.cacheKeyResolver = b.cacheKeyResolver;
    }

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private boolean useBuiltInCache = false;
        private File cacheDir = null;
        private long cacheMaxBytes = DEFAULT_CACHE_MAX_BYTES;
        private CacheKeyResolver cacheKeyResolver = null;

        public Builder setUseBuiltInCache(boolean use) {
            this.useBuiltInCache = use;
            return this;
        }

        public Builder setCacheDir(@Nullable File dir) {
            this.cacheDir = dir;
            return this;
        }

        public Builder setCacheMaxBytes(long bytes) {
            this.cacheMaxBytes = bytes;
            return this;
        }

        public Builder setCacheKeyResolver(@Nullable CacheKeyResolver resolver) {
            this.cacheKeyResolver = resolver;
            return this;
        }

        public CacheConfig build() {
            return new CacheConfig(this);
        }
    }
}

File: dkplayer-players/media3Exo/src/main/java/xyz/doikki/videoplayer/media3/CacheConfig.kt
package xyz.doikki.videoplayer.media3

import java.io.File

class CacheConfig private constructor(
    val useBuiltInCache: Boolean,
    val cacheDir: File?,
    val cacheMaxBytes: Long,
    val cacheKeyResolver: CacheKeyResolver?
) {
    class Builder {
        private var useBuiltInCache = false
        private var cacheDir: File? = null
        private var cacheMaxBytes = DEFAULT_CACHE_MAX_BYTES
        private var cacheKeyResolver: CacheKeyResolver? = null

        fun setUseBuiltInCache(use: Boolean) = apply { useBuiltInCache = use }
        fun setCacheDir(dir: File?) = apply { this.cacheDir = dir }
        fun setCacheMaxBytes(bytes: Long) = apply { cacheMaxBytes = bytes }
        fun setCacheKeyResolver(resolver: CacheKeyResolver?) = apply { cacheKeyResolver = resolver }
        fun build() = CacheConfig(useBuiltInCache, cacheDir, cacheMaxBytes, cacheKeyResolver)
    }

    companion object {
        const val DEFAULT_CACHE_MAX_BYTES = 512L * 1024 * 1024
        @JvmStatic
        fun builder() = Builder()
    }
}
